libparsec/crates/platform_storage/src/web/model.rs

```diff
@@ -0,0 +1,195 @@
+}
+
+pub(super) async fn initialize_model_if_needed(
+    mut db_req: OpenDbRequest,
+) -> anyhow::Result<IdbDatabase> {
+    db_req.set_on_upgrade_needed(Some(|evt: &IdbVersionChangeEvent| {
+        if let None = evt.db().object_store_names().find(|n| n == Vlob::STORE) {
+            let store = evt.db().create_object_store_with_params(
+                Vlob::STORE,
+                IdbObjectStoreParameters::default().auto_increment(true),
+            )?;
+            store.create_index(Vlob::INDEX_VLOB_ID, &IdbKeyPath::str(Vlob::INDEX_VLOB_ID))?;
+        }
+        if let None = evt
+            .db()
+            .object_store_names()
+            .find(|n| n == RealmCheckpoint::STORE)
+        {
+            let store = evt.db().create_object_store_with_params(
+                RealmCheckpoint::STORE,
+                IdbObjectStoreParameters::default().auto_increment(true),
+            )?;
+            store.create_index(
+                RealmCheckpoint::INDEX_ID,
+                &IdbKeyPath::str(RealmCheckpoint::INDEX_ID),
+            )?;
+        }
+        if let None = evt
+            .db()
+            .object_store_names()
+            .find(|n| n == PreventSyncPattern::STORE)
+        {
+            let store = evt.db().create_object_store_with_params(
+                PreventSyncPattern::STORE,
+                IdbObjectStoreParameters::default().auto_increment(true),
+            )?;
+            store.create_index(
+                PreventSyncPattern::INDEX_ID,
+                &IdbKeyPath::str(PreventSyncPattern::INDEX_ID),
+            )?;
+        }
+        if let None = evt.db().object_store_names().find(|n| n == Chunk::STORE) {
+            let store = evt.db().create_object_store_with_params(
+                Chunk::STORE,
+                IdbObjectStoreParameters::default().auto_increment(true),
+            )?;
+            store.create_index(
+                Chunk::INDEX_CHUNK_ID,
+                &IdbKeyPath::str(Chunk::INDEX_CHUNK_ID),
+            )?;
+        }
+        if let None = evt
+            .db()
+            .object_store_names()
+            .find(|n| n == Remanence::STORE)
+        {
+            let store = evt.db().create_object_store_with_params(
+                Remanence::STORE,
+                IdbObjectStoreParameters::default().auto_increment(true),
+            )?;
+            store.create_index(Remanence::INDEX_ID, &IdbKeyPath::str(Remanence::INDEX_ID))?;
+        }
+        if let None = evt
+            .db()
+            .object_store_names()
+            .find(|n| n == Certificate::STORE)
+        {
+            let store = evt.db().create_object_store_with_params(
+                Certificate::STORE,
+                IdbObjectStoreParameters::default().auto_increment(true),
+            )?;
+            store.create_index(
+                Certificate::INDEX_CERTIFICATE_TYPE,
+                &IdbKeyPath::str(Certificate::INDEX_CERTIFICATE_TYPE),
+            )?;
+        }
```

Comment:
Can you split and do a check per store ?

something like

```rust
create_certificate_store(evt);
create_remanence_store(evt);
```

Reply:
I will make methods